src/userSurvey.js

```diff
@@ -144,34 +172,50 @@ function setPopupToNoPromptPage() {
  */
 export async function setSurvey(options) {
-    if(createdSurvey) {
-        throw new Error("userSurvey only supports one survey at present.");
+    initializeStorage();
+
+    let currentSurveyDetails = await storageSpace.get("currentSurveyDetails");
+    let surveyDetails;
+
+    if (!currentSurveyDetails) {
+        // If there's a call to setSurvey and there's no survey in storage,
+        // save the parameters in storage and carry out the survey.
+        surveyDetails = options;
+        await storageSpace.set("currentSurveyDetails", options);
+    } else if (currentSurveyDetails.surveyName === options.surveyName) {
+        // If there's a call to setSurvey with the same survey name as the
+        // survey in storage, ignore options and load all the survey attributes
+        // from storage, and continue with the survey.
+        surveyDetails = await storageSpace.get("currentSurveyDetails");
+    } else {
+        // If there's a call to setSurvey with a different survey name from
+        // the survey in storage, throw an error, because two surveys are
+        // now set at the same time.
+        throw new Error("userSurvey only supports one survey at a time. Complete the survey that has previously been set.");
     }
-    createdSurvey = true;
+
+
 
     const currentTime = Date.now();
-    surveyUrl = options.surveyUrl;
-    reminderIconUrl = browser.runtime.getURL(options.reminderIcon);
-    reminderInterval = options.reminderInterval;
-    reminderTitle = options.reminderTitle;
-    reminderMessage = options.reminderMessage;
+    surveyUrl = surveyDetails.surveyUrl;
+    reminderIconUrl = browser.runtime.getURL(surveyDetails.reminderIcon);
+    reminderInterval = surveyDetails.reminderInterval;
+    reminderTitle = surveyDetails.reminderTitle;
+    reminderMessage = surveyDetails.reminderMessage;
     browser.storage.local.set({
-        "webScience.userSurvey.popupPromptMessage": options.popupPromptMessage
+        "webScience.userSurvey.popupPromptMessage": surveyDetails.popupPromptMessage
     });
     browser.storage.local.set({
-        "webScience.userSurvey.popupNoPromptMessage": options.popupNoPromptMessage
+        "webScience.userSurvey.popupNoPromptMessage": surveyDetails.popupNoPromptMessage
     });
 
-    storageSpace = storage.createKeyValueStorage("webScience.userSurvey");
-    /* Check when we last asked the user to do the survey. If it's null,
-     * we've never asked, which means the extension just got installed.
-     * Open a tab with the survey, and save this time as the most recent
-     * request for participation.
-     */
+    // Check when we last asked the user to do the survey. If it's null,
+    // we've never asked, which means the extension just got installed.
+    // Open a tab with the survey, and save this time as the most recent
+    // request for participation.
     let lastSurveyRequest = await storageSpace.get("lastSurveyRequest");
```

Comment:
```suggestion
    lastSurveyRequest = await storageSpace.get("lastSurveyRequest");
```
This is a bug I introduced. `lastSurveyRequest` should not be redeclared in function scope.